src/index.ts

```diff
@@ -54,25 +54,68 @@ const start = async (args: { config: string }): Promise<void> => {
-  log.debug(`Account ${account.address} has free balance of: ${dm} ${api.registry.chainTokens}`)
+  const base = 10**Number(decimals);
+  const dm = Number(balance.free) / base;
+  log.debug(`Account ${account.address} has free balance of: ${dm} ${api.registry.chainTokens}`);
 
   const existentialDeposit = Number(api.consts.balances.existentialDeposit) / base;
   // Sending funds to addresses
-  const share = parseFloat(config.rewardsDestination.mainDestinationShare) / 100;
-  log.debug(`Share split: ${share * 100}%`)
-  const mainBalance = (Number(balance.free)/ base) *share;
-  const dustBalance = (Number(balance.free)/ base) * (1-share);
-  log.debug(`Will send ${mainBalance} ${api.registry.chainTokens} to ${config.rewardsDestination.mainDestinationAddress}`)
-  log.debug(`Will send ${dustBalance} ${api.registry.chainTokens} (minus fees) to ${config.rewardsDestination.dustDestinationAddress}`)
-  if(mainBalance < existentialDeposit || dustBalance < existentialDeposit) {
-    log.debug(`Warning, sending less than the Existencial Deposit! If target address has less than that it will be lost! Press Ctrl+C now to stop!`)
-    await delay(10000);
+  const share = parseFloat(config.rewardsDestination.primaryDestinationShare) / 100;
+  log.debug(`Share split: ${share * 100}%`);
+
+  // We need to handle the ED of three addresses. First, we need to make sure that the balance of the Account we send from will not fall
+  // below the ED after sending it. Afterwards, we need to make sure that funds do not get lost on the primaryDestinationAddress and 
+  // secondaryDestinationAddress. There are two options here. Either the available_funds * share and available_funds * (1 - share) are both
+  // above the ED or, if they are below the ED, both primaryDestinationAddress and secondaryDestinationAddress are already having more than the ED.
+  // We only want to execute the transactions if no funds are lost anywhere, this should work without user interaction.
+
+  // Get primary and secondary account balances.
+  const { data: balance_primaryDestination } = await api.query.system.account(config.rewardsDestination.primaryDestinationAddress);
+  const { data: balance_secondaryDestination } = await api.query.system.account(config.rewardsDestination.secondaryDestinationAddress);
+
+  var primaryDestinationOverED;
+  var secondaryDestinationOverED;
+
+  // We check whether the two destination addresses have the ED.
+  if((Number(balance_primaryDestination.free) / base) >= existentialDeposit){
+    primaryDestinationOverED = true;
+  } else {
+    primaryDestinationOverED = false;
+  }
+
+  if((Number(balance_secondaryDestination.free) / base) >= existentialDeposit){
+    secondaryDestinationOverED = true;
+  } else {
+    secondaryDestinationOverED = false;
+  }
+
+  // We make sure that the remaining funds are exactly the ED.
+  // TODO: We need to reserve funds for two upcoming transactions. How do we do this? Right now, I just heuristically reserve 
```

Comment:
You mean the fees? Fee estimates can be calculated like this: https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-estimate-the-transaction-fees

Reply:
Yes, we should reserve tx-fees for the upcoming two transactions (since we want to send everything such that only ED is left on the origin address)